pages/index.vue

```diff
@@ -48,27 +46,21 @@ export default {
       .finally(() => {
         this.loading = false;
         this.isLastPage =
-          this.filters.offset + this.filters.limit >= this.totalCharacters;
+          this.$route.params.offset + this.$route.params.limit >=
+          this.totalCharacters;
       });
   },
 
-  watch: {
-    filters: {
-      deep: true,
-      handler() {
-        this.$fetch();
-      },
-    },
-  },
-
   methods: {
     appendNextPage() {
-      this.filters.offset += 20;
+      this.$route.params.offset += 20;
+      this.$fetch();
     },
 
     debounceSearch: _.debounce(function (e) {
-      this.filters.offset = 0;
-      this.filters.nameStartsWith = e || undefined;
+      this.$route.params.offset = 0;
+      this.$route.params.nameStartsWith = e || undefined;
```

Comment:
Da forma atual, o sistema não está atualizando a URL.

Teste da seguinte maneira:
```suggestion
      this.$router.push({path: this.$route.path, query: { nameStartsWith: e || undefined }})
```

Sugiro dar uma olhada: https://github.com/nuxt/nuxt.js/issues/1101